fpga/src/main/scala/arty/Configs.scala

```diff
@@ -40,6 +41,23 @@ class WithDefaultPeripherals extends Config((site, here, up) => {
     p.copy(hang = 0x10000, contentFileName = s"./fpga/src/main/resources/arty/xip/build/xip.bin")
   }
 })
+
+class ArtyGPIOPeripheral extends Config((site,here,up) => {
+    case PeripheryGPIOKey => List(
+    GPIOParams(
+      address = 0x10012000,
+      width = 32,
```

Comment:
Instead of adding this key which sets everything, lets add new classes
`class WithGPIOWidth(x: Int)`
and
`class WithGPIOIncludeIOF(iof: Boolean)` to the chipyard ConfigFragments.scala. These classes should read and override the base values set in `PeripheryGPIOKey`.